marimbabot_behavior/scripts/behavior_node_dummy.py

```diff
@@ -0,0 +1,27 @@
+# This node is a dummy node that will be used to test the behavior tree
+# It emulates voice commands and publishes to the behavior node
+# It emulates vision information and publishes to the behavior node
+
+import rospy
+from std_msgs.msg import String
+
+
+class DummyBehaviorNode:
+    def __init__(self):
+        rospy.init_node('dummy_behavior_node', anonymous=True)
+        self.command_pub = rospy.Publisher('speech_node/command', String, queue_size=10)
+        self.recognized_notes_pub = rospy.Publisher('vision_node/recognized_notes', String, queue_size=10)
+        self.main_loop()
+
+    def main_loop(self):
+        """
+        Wait for keyboard input, concatenate the input into a string, and send it to the server once the user presses enter.
+        """
+        while not rospy.is_shutdown():
+            # get input from user and create goal
+            input_str = input("Enter a command (e.g. 'marimbabot read'): ")
+            recognized_notes_str = input("Enter recognized notes (default 'c4 d4 e4'): ", )
```

Comment:
New vision input and commands are coupled together. You would not want to update the vision data if you only want to send a `marimbabot slower`. I personally do not really see the benefits over a simple `rostopic pub ...` in the current state. Maybe if we would add numbered choices for commands, a simple gui or something like that.

Reply:
@yunlong-wang-cn

Reply:
@yunlong-wang-cn 
If we don't merge this branch you could still use this locally if it is handy to you  😅